src/interpret.c

```diff
@@ -4194,11 +4194,11 @@ class Interpreter : public Visitor
                 Expression *aggr2 = se->e1;
                 dinteger_t srclower = se->lwr->toInteger();
                 dinteger_t srcupper = se->upr->toInteger();
-                bool wantCopy = (newval->type->toBasetype()->baseElemOf()->ty == Tstruct);
+                bool wantCopy = (newval->type->toBasetype()->nextOf()->baseElemOf()->ty == Tstruct);
```

Comment:
When I fixed issue 14024, I added this code block. But there is some implementation bugs, and those were detected by this change.